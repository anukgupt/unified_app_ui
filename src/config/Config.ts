
export const msalConfig = {
    appId: '86570ef4-5443-4de9-bb47-1542755d3565',
    authority: 'https://login.microsoftonline.com/',
    // redirectUri: 'http://localhost:3000/' ,
    redirectUri: 'https://unifiedappui.azurewebsites.net',
    scopes: ["User.Read"],
    azureScopes: [
        'https://management.azure.com/user_impersonation',
        'User.Read'
    ],
    azureApiScopes: [
        'https://management.azure.com/user_impersonation'
    ]
};

export const basicURLS = {
    clientservicepermissionsSaveURL: "https://localhost:44384/api/clientservicepermissions",
    fetchSubscriptionsURL: "https://management.azure.com/subscriptions?api-version=2020-01-01",
    fecthTenantsURL: "https://management.azure.com/tenants?api-version=2020-01-01",
    installationMappingURL: "https://localhost:44384/api/installationmapping",
    gitHubUserAuthURL: "https://github.com/login/oauth/authorize?client_id=Iv1.6bd7888b1e58ccad&redirect_uri=https://localhost:44384/api/usertoken&state="
}
