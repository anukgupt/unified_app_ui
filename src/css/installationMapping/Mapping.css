html, body{
    padding: 0;
    margin: 0;
}
html {
    height: 100%;
}
body {  
    font-size: 16px;
    box-sizing: border-box;
    height: 100%;
    width: 100%;
    min-height: 100%;
    color: #323130;
    background: #FFFFFF;
}
.full-size {
    display: flex;
    flex-grow: 1;
    overflow: hidden;
} 
.outer {  
    box-shadow: 0 2px 3px rgba(0, 0, 0, 0.55);
    border: 1px solid rgba(0, 0, 0, 0.4);
    width: 480px;
    margin-top: 120px;
    margin-left: auto;
    margin-right: auto;
}
.inner {
    margin-right: 50px;
    font-family: 'Segoe UI';
}
.heading-container {
    display: flex;
    align-items: center;
}
.heading-image {
    width: 36px;
    margin-left: 58px;
    margin-top: 52px;
    height: 36px;
}
.heading-text {
    width: 116px;
    margin-left: 15px;
    margin-top: 68px;
}
.heading-app-text {
    height: 33px;
    margin-left: 55px;
    margin-top: 23px;
    font-size: 24px;
}
.form-items {
    display: flex;
    flex-direction: column;
    margin-left: 56px;
}
.formitem-tenant {
    margin-top: 30px;
    font-size: 14px;
}
.formitem-dropdown {
    align-items: center;
    display: flex;
    text-overflow: ellipsis;
    white-space: nowrap;
    width: 370px;
    margin-top: 8px;
}
.formitem-button-value {
    float: left;
    font-weight: 400;
    font-size: 13px;
    color: rgba(0, 0, 0, 0.9);
    margin-left: 5px;
}
.formitem-dropdown-arrow {
    float: right;
    height: 12px;
    width: 12px;
    margin-top: 5px;
    margin-right: 5px;
}
.tenant-displayname {
    font-size: 13px;
    color: rgba(0, 0, 0, 0.9);
}
.formitem-subscription {
    font-size: 14px;
    margin-top: 30px;
}
.input-checkbox-label {
    font-size: 13px;
    line-height: 18px;
    color: rgba(0, 0, 0, 0.9);
    font-weight: 400;
}
.input-checkbox {
    width: 16px;
    height: 16px;
    margin-right: 8px;
}
.selected-subscriptions-list {
    margin-top: 20px;
    border-bottom: 1px solid #e1e4e8;
    width: 370px;
}
.heading-subscriptions-selected {
    font-size: 12px;
}
.subs-list {
    margin-top: 10px;
    font-size: 13px;
    margin-bottom: 10px;
}
.overflow {
    height: 80px;
    overflow-y: auto;
}
.subscription-item {
    display: flex;
    flex-direction: row;
    margin-top: 5px;
    align-items: center;
}
.subscription-item-name {
    width: 90%;
}
.subscription-remove-button {
    border: 0;
    box-shadow: none;
    background-color: #ffffff;
}
.remove-button {
    float: right;
    font-size: 13px;
    font-weight: 400;
    color: #0078D4; 
}
.link {
    color: #005a9e;
    cursor: pointer;
    outline: transparent;
    text-decoration: none;
}
.subscription-link {
    margin-top: 5px;
    font-size: 12px;
    line-height: 14px;
}
.description-link {
    margin-top: 30px;
    margin-right: 49px;
    height: 34px;
    font-size: 12px;
    line-height: 14px;
}
.text-right {
    text-align: right;
}
.bolt-button {
    background-color: #0078d4;
    margin-top: 40px;
    font-size: 14px;
    text-align: center;
    margin-bottom: 35px;
    color: #ffffff;
    height: 35px;
    width: 70px;
    border: 0;
}
input, select, textarea, button {
    color: inherit;
    font-family: inherit;
    font-size: inherit;
}  
* {
    box-sizing: inherit;
}
.errorMessage, .errMessage {
    color: #A4262C;
}
.successMessage, .errorMessage {
    padding-left: 55px;
    padding-top: 15px;
    display: table;
}
.successMessage {
    color: green;
}
.errMessage {
    display: table;
}
.success-img, .error-img, .err-img {
    position: relative;
    top: 5px;
    display: table-cell;
}
.success-msg, .error-msg, .err-msg {
    display: table-cell;
    padding-left: 10px;
    font-size: 13px;
}