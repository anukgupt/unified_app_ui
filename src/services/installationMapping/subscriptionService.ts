import { basicURLS } from "../../config/Config";

export interface ISubscription {
  subscriptionId: string,
  displayName: string
}

export async function getAllSubscriptions(accessToken: string) {
  var url = basicURLS.fetchSubscriptionsURL;
  return getSubscriptions(url, accessToken).then(subscriptionResponse => {
    var subscriptions: Array<ISubscription> = [];
    subscriptions = subscriptions.concat(subscriptionResponse.value);
    while (subscriptionResponse.nextLink) {
      return getSubscriptions(subscriptionResponse.nextLink, accessToken).then(sub => {
        subscriptions = subscriptions.concat(sub);
        return subscriptions;
      });
    }
    return subscriptions;
  });
}

export async function getSubscriptions(url: string, accessToken: string) {
  var bearer = 'Bearer ' + accessToken;
  return fetch(url, {
    method: 'GET',
    headers: {
      'Authorization': bearer
    }
  }).then(responseJson => {
    return responseJson.json().then(data => {
      return data
    });
  }).catch(error => {
    throw error;
  });
}