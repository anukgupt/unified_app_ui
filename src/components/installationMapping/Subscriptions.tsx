import React from 'react';
import { msalConfig } from '../../config/Config';
import { getAllSubscriptions } from '../../services/installationMapping/subscriptionService';
import { Button, SelectMenu } from '@primer/components';
import down from '../../images/downarrow.png';
import * as Constants from "../../Constants";
import '../../css/installationMapping/Mapping.css';
import redcross from '../../images/redcross.png';

export interface ISubscriptionProps {
  getAccessToken: any,
  setError: any,
  tenantId: string,
  setSubscriptionIds: any,
  setAccessToken: any,
  setErrorOnFetchingSubscriptions: any,
  errorOnFetchingSubscriptions: boolean,
  errorOnFetchingTenants: boolean
}

interface SubscriptionParameters {
  subscriptions: any[];
  filteredSubscriptions: any[];
  finalSubscriptionList: any[];
  currentTenantId: string;
  subscriptionRadioButtonValue: string;
  searchValueForSubscriptions: string;
}

interface SubscriptionState {
  subscriptionParameters: SubscriptionParameters;
}

export class Subscription extends React.Component<ISubscriptionProps, SubscriptionState> {
  constructor(props: any) {
    super(props);
    let subscriptionParameters: SubscriptionParameters = {
      subscriptions: [],
      filteredSubscriptions: [],
      finalSubscriptionList: [],
      currentTenantId: '',
      subscriptionRadioButtonValue: Constants.AllSubscriptionsRadioButtonValue,
      searchValueForSubscriptions: ""
    }
    this.state = {
      subscriptionParameters: subscriptionParameters
    };
  }

  setSubscriptionParameters(subscriptionParameters: SubscriptionParameters) {
    this.setState({ subscriptionParameters: subscriptionParameters });
  }

  async componentDidUpdate() {
    try {
      let subscriptionParameters = this.state.subscriptionParameters;
      if (this.props.tenantId !== this.state.subscriptionParameters.currentTenantId) {
        if (this.props.tenantId) {
          var accessToken = await this.props.getAccessToken(this.props.tenantId, msalConfig.azureApiScopes);
          var subscriptions = await getAllSubscriptions(accessToken);
          subscriptionParameters.subscriptions = subscriptions;
          subscriptionParameters.currentTenantId = this.props.tenantId;
          subscriptionParameters.filteredSubscriptions = subscriptions;
        } else {
          subscriptionParameters.subscriptions = [];
          subscriptionParameters.currentTenantId = '';
          subscriptionParameters.filteredSubscriptions = [];
        }
        subscriptionParameters.finalSubscriptionList = [];
        subscriptionParameters.subscriptionRadioButtonValue = Constants.AllSubscriptionsRadioButtonValue;
        subscriptionParameters.searchValueForSubscriptions = "";
        this.props.setSubscriptionIds(subscriptionParameters.subscriptions);
        this.setSubscriptionParameters(subscriptionParameters);
        this.props.setAccessToken(accessToken);
      }
    }
    catch (err) {
      this.props.setErrorOnFetchingSubscriptions(true);
      this.props.setError('ERROR', JSON.stringify(err));
    }
  }

  render() {
    let isPresentInFinalList: boolean = false;
    let showSubscriptionFilter: boolean = false;
    if (this.state.subscriptionParameters.filteredSubscriptions.length > 0) {
      showSubscriptionFilter = true;
    }
    let options: any = [];
    this.state.subscriptionParameters.filteredSubscriptions && this.state.subscriptionParameters.filteredSubscriptions.length > 0 &&
      this.state.subscriptionParameters.filteredSubscriptions.map(subs => {
        isPresentInFinalList = false;
        this.state.subscriptionParameters.finalSubscriptionList.map(finalSub => {
          if (subs.displayName === finalSub.displayName) {
            isPresentInFinalList = true;
          }
        });
        options.push(<SelectMenu.Item key={subs.subscriptionId} className="formitem-selectmenu-item">
          <input id={subs.displayName} className="input-checkbox" type="checkbox" checked={isPresentInFinalList} onClick={(event) => this.onSelectingSubscription(event)} />
          <label htmlFor={subs.displayName} className="input-checkbox-label">{subs.displayName}</label>
        </SelectMenu.Item>);
      });
    let finalList: any = [];
    this.state.subscriptionParameters.finalSubscriptionList.map(subs => {
      finalList.push(
        <div key={subs.subscriptionId} className="subscription-item">
          <span className="subscription-item-name">{subs.displayName}</span>
          <button className="subscription-remove-button" type="button" aria-label="remove-button" onClick={() => {
            this.onRemovalOfSubscription(subs.displayName)
          }}>
            <span className="remove-button">{Constants.RemoveButtonText}</span>
          </button>
        </div>
      );
    });
    let subList = <div className="subs-list">
      {
        finalList
      }
    </div>
    if (this.state.subscriptionParameters.finalSubscriptionList.length > 3) {
      subList = <div className="subs-list overflow">
        {
          finalList
        }
      </div>
    }
    var formitemSubscription = "formitem-subscription";
    if (this.props.errorOnFetchingTenants) {
      formitemSubscription = "formitem-subscription margin-top";
    }
    return (
      <div className={formitemSubscription}>
        <span>{Constants.SelectAzureSubscriptionLabel}</span>
        <SelectMenu className="formitem-dropdown">
          <Button className="formitem-button" as="summary">
            <span className="formitem-button-value">{Constants.SelectSubscriptionsPlaceholder}</span>
            <img className="formitem-dropdown-arrow" alt="" src={down} />
          </Button>
          {
            showSubscriptionFilter &&
            <SelectMenu.Modal width="370px">
              <SelectMenu.Filter height="18px" className="selectmenu-filter" onChange={(event: any) => { this.onFilteringSubscription(event.target.value) }} placeholder={Constants.DropdownFilterAriaLabel} value={this.state.subscriptionParameters.searchValueForSubscriptions} aria-label={Constants.DropdownFilterAriaLabel} />
              <SelectMenu.Header className="formitem-header">{Constants.AllSubscriptionsRadioButtonValue}</SelectMenu.Header>
              <SelectMenu.List className="all-subscriptions-list">
                <SelectMenu.Item key={Constants.AllSubscriptionsRadioButtonValue} className="formitem-selectmenu-item">
                  <input id="input-checkbox-all" className="input-checkbox" type="checkbox" checked={this.state.subscriptionParameters.subscriptionRadioButtonValue == Constants.AllSubscriptionsRadioButtonValue} onClick={() => this.onAllSubscriptionsCheckBox()} value={Constants.AllSubscriptionsRadioButtonNote} />
                  <label htmlFor="input-checkbox-all" className="input-checkbox-label">{Constants.AllSubscriptionsRadioButtonNote}</label>
                </SelectMenu.Item>
              </SelectMenu.List>
              <SelectMenu.Header className="formitem-header">{Constants.SelectSubscriptionsRadioButtonValue}</SelectMenu.Header>
              <SelectMenu.List className="select-subscriptions-list">
                {
                  options
                }
              </SelectMenu.List>
            </SelectMenu.Modal>
          }
          {
            !showSubscriptionFilter &&
            <SelectMenu.Modal width="370px">
              <SelectMenu.Filter height="18px" className="selectmenu-filter" onChange={(event: any) => { this.onFilteringSubscription(event.target.value) }} placeholder={Constants.DropdownFilterAriaLabel} value={this.state.subscriptionParameters.searchValueForSubscriptions} aria-label={Constants.DropdownFilterAriaLabel} />
              <SelectMenu.List>
              </SelectMenu.List>
            </SelectMenu.Modal>
          }
        </SelectMenu>
        {
          this.props.errorOnFetchingSubscriptions &&
          <div className="errMessage">
            <img className="err-img" src={redcross} width="20px" height="20px" />
            <span className="err-msg">{Constants.ErrorOnFetchingSubscriptionsMessage}</span>
          </div>
        }
        {
          this.state.subscriptionParameters.finalSubscriptionList && this.state.subscriptionParameters.finalSubscriptionList.length > 0 &&
          <div className="selected-subscriptions-list">
            <span className="heading-subscriptions-selected">Selected {this.state.subscriptionParameters.finalSubscriptionList.length} subscriptions</span>
            {
              subList
            }
          </div>
        }
      </div >
    )
  }
  filterSubscriptions(searchValue: string, subscriptionParameters: SubscriptionParameters): SubscriptionParameters {
    let filteredSubs: any[] = [];
    subscriptionParameters.subscriptions.filter(sub => {
      if (sub.displayName.toString().toLowerCase().indexOf(searchValue.toString().toLowerCase()) !== -1)
        filteredSubs.push(sub);
    });
    subscriptionParameters.filteredSubscriptions = filteredSubs;
    subscriptionParameters.searchValueForSubscriptions = searchValue;
    return subscriptionParameters;
  }

  onFilteringSubscription(searchValue: string) {
    let subscriptionParameters = this.filterSubscriptions(searchValue, this.state.subscriptionParameters);
    this.setSubscriptionParameters(subscriptionParameters);
  }

  onSelectingSubscription(event: any | React.MouseEvent<HTMLInputElement, MouseEvent>) {
    let subscriptionParameters = this.state.subscriptionParameters;
    let subs = subscriptionParameters.finalSubscriptionList;
    let tobeAddedSubscription = subscriptionParameters.filteredSubscriptions.filter(sub => {
      return (sub.displayName.indexOf(event.currentTarget.id) !== -1);
    });
    if (!event.target.checked) {
      this.onRemovalOfSubscription(event.currentTarget.id);
    }
    else {
      subs[subscriptionParameters.finalSubscriptionList.length] = tobeAddedSubscription[0];
      subscriptionParameters.finalSubscriptionList = subs;
      subscriptionParameters.subscriptionRadioButtonValue = Constants.SelectSubscriptionsRadioButtonValue;
      this.props.setSubscriptionIds(subscriptionParameters.finalSubscriptionList);
      this.setSubscriptionParameters(subscriptionParameters);
    }
  }

  onAllSubscriptionsCheckBox() {
    let subscriptionParameters = this.state.subscriptionParameters;
    if (subscriptionParameters.subscriptionRadioButtonValue == Constants.AllSubscriptionsRadioButtonValue) {
      subscriptionParameters.subscriptionRadioButtonValue = Constants.SelectSubscriptionsRadioButtonValue;
      this.props.setSubscriptionIds([]);
    }
    else {
      subscriptionParameters.subscriptionRadioButtonValue = Constants.AllSubscriptionsRadioButtonValue;
      subscriptionParameters.finalSubscriptionList = [];
      this.props.setSubscriptionIds(subscriptionParameters.subscriptions);
    }
    this.setSubscriptionParameters(subscriptionParameters);
  }

  onRemovalOfSubscription(subscriptionDisplayName: string) {
    let subscriptionParameters = this.state.subscriptionParameters;
    let subs = subscriptionParameters.finalSubscriptionList;
    let indexToBeDeleted = -1;
    subs.map((subs, index) => {
      if (subs.displayName === subscriptionDisplayName) {
        indexToBeDeleted = index;
      }
    });
    if (indexToBeDeleted !== -1) {
      subs.splice(indexToBeDeleted, 1);
      subscriptionParameters.finalSubscriptionList = subs;
      this.props.setSubscriptionIds(subscriptionParameters.finalSubscriptionList);
      if (subscriptionParameters.finalSubscriptionList.length == 0) {
        subscriptionParameters.subscriptionRadioButtonValue = Constants.AllSubscriptionsRadioButtonValue;
        this.props.setSubscriptionIds(subscriptionParameters.subscriptions);
      }
      this.setSubscriptionParameters(subscriptionParameters);
    }
  }
}

export default Subscription;