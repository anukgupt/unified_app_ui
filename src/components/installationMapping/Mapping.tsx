import * as React from 'react';
import withAuthProvider from '../../provider/AuthProviderForInstallationMapping';
import '../../css/installationMapping/Mapping.css';
import ClipLoader from "react-spinners/ClipLoader";
import Tenants, { ITenantsProps } from './Tenants';
import { IMappingInput, saveMapping } from "../../services/installationMapping/mappingService";
import azurelogo from '../../images/azurelogo.png';
import * as Constants from '../../Constants';
import Subscriptions, { ISubscriptionProps } from './Subscriptions';
import { msalConfig, basicURLS } from '../../config/Config';
import { getTenants } from '../../services/installationMapping/tenantService';
import background from '../../images/background.png';
import greentick_checkmark from '../../images/greentick_checkmark.png';
import redcross from '../../images/redcross.png';

export interface MappingStateObject {
    successMessage: string,
    error: boolean,
    errorMessage: string,
    loading: boolean,
    installationId: string,
    ghstate: string,
    subscriptionIds: string[],
    selectedTenant: {
        id: string,
        name: string
    },
    tenants: {
        value: any[]
    },
    errorOnFetchingTenants: boolean,
    errorOnFetchingSubscriptions: boolean,
    accessToken: string
}

interface MappingState {
    mappingStateObject: MappingStateObject
}

class Mapping extends React.Component<any, MappingState> {
    constructor(props: any) {
        super(props);
        let mappingStateObject: MappingStateObject = {
            successMessage: "",
            error: false,
            errorMessage: "",
            loading: false,
            installationId: this.getQueryParam("installation_id"),
            ghstate: this.getQueryParam("state"),
            subscriptionIds: [],
            selectedTenant: {
                id: "",
                name: ""
            },
            tenants: {
                value: []
            },
            errorOnFetchingTenants: false,
            errorOnFetchingSubscriptions: false,
            accessToken: ""
        }
        this.state = {
            mappingStateObject: mappingStateObject
        };
    }

    private getQueryParam(paramName: string) {
        let param = "";
        let url: URL = new URL(window.location.href);
        const queryString = new URLSearchParams(url.search);
        if (queryString.has(paramName)) {
            param = queryString.get(paramName) || param;
        }

        return param;
    }

    async componentDidMount() {
        try {
            let accessToken = await this.props.getAccessToken('', msalConfig.azureScopes);
            let tenants = await getTenants(accessToken);
            this.setTenants(tenants);
        }
        catch (err) {
            var mappingStateObject = this.state.mappingStateObject;
            mappingStateObject.errorOnFetchingTenants = true;
            this.props.setError('ERROR', JSON.stringify(err));
        }
    }

    setMappingState = (mappingStateObject: MappingStateObject) => {
        this.setState({ mappingStateObject: mappingStateObject });
    }

    private setAccessToken(accessToken: string) {
        let mappingStateObject = this.state.mappingStateObject;
        mappingStateObject.accessToken = accessToken;
        this.setMappingState(mappingStateObject);
    }

    private setLoading(loading: boolean) {
        let mappingStateObject = this.state.mappingStateObject;
        mappingStateObject.loading = loading;
        this.setMappingState(mappingStateObject);
    }

    private setInstallationSuccess() {
        let mappingStateObject = this.state.mappingStateObject;
        mappingStateObject.successMessage = Constants.InstallationMappingSuccessfullMessage;
        mappingStateObject.loading = false;
        this.setMappingState(mappingStateObject);
    }

    private setInstallationError(error: any) {
        let mappingStateObject = this.state.mappingStateObject;
        if (error instanceof Error) {
            mappingStateObject.errorMessage = error.message;
        }
        mappingStateObject.error = true;
        mappingStateObject.loading = false;
        this.setMappingState(mappingStateObject);
    }

    private setTenants(tenants: any) {
        let mappingStateObject = this.state.mappingStateObject;
        mappingStateObject.tenants = tenants;
        this.setMappingState(mappingStateObject);
    }

    private setSelectedTenant(tenant: any) {
        let mappingStateObject = this.state.mappingStateObject;
        mappingStateObject.selectedTenant = {
            id: tenant.tenantId,
            name: tenant.displayName
        };
        this.setMappingState(mappingStateObject);
    }

    private setSubscriptionIds(subscriptionsIds: any) {
        let mappingStateObject = this.state.mappingStateObject;
        mappingStateObject.subscriptionIds = subscriptionsIds;
        this.setMappingState(mappingStateObject);
    }

    private setErrorOnFetchingSubscriptions(res: boolean) {
        let mappingStateObject = this.state.mappingStateObject;
        mappingStateObject.errorOnFetchingSubscriptions = res;
        this.setMappingState(mappingStateObject);
    }

    private saveInstallationMapping() {
        this.setLoading(true);
        let mappingInput: IMappingInput = {
            installationId: this.state.mappingStateObject.installationId,
            tenantId: this.state.mappingStateObject.selectedTenant.id,
            subscriptionIds: this.state.mappingStateObject.subscriptionIds
        };
        saveMapping(mappingInput, this.state.mappingStateObject.accessToken).then(result => {
            this.setInstallationSuccess();

            if (this.state.mappingStateObject.ghstate !== "") {
                this.redirectToUserAuth(this.state.mappingStateObject.ghstate);
            }
        }).catch(ex => {
            this.setInstallationError(ex);
        });
    }

    private redirectToUserAuth(ghstate: string) {
        window.location.href = basicURLS.gitHubUserAuthURL + ghstate;
    }

    render() {
        const backgourndImageUrl = `url(${background})`;
        let tenantProps: ITenantsProps = {
            mappingStateObject: this.state.mappingStateObject,
            setTenantId: this.setSelectedTenant.bind(this)
        }

        let subscriptionProps: ISubscriptionProps = {
            getAccessToken: this.props.getAccessToken,
            setError: this.props.setError,
            tenantId: this.state.mappingStateObject.selectedTenant.id,
            setSubscriptionIds: this.setSubscriptionIds.bind(this),
            setAccessToken: this.setAccessToken.bind(this),
            setErrorOnFetchingSubscriptions: this.setErrorOnFetchingSubscriptions.bind(this),
            errorOnFetchingSubscriptions: this.state.mappingStateObject.errorOnFetchingSubscriptions,
            errorOnFetchingTenants: this.state.mappingStateObject.errorOnFetchingTenants
        };
        return (
            <div className="full-size">
                <div
                    className="background-image"
                    style={backgourndImageUrl ? {
                        backgroundImage: backgourndImageUrl
                    } : undefined}
                >
                    <div className="outer">
                        <div className="inner">
                            <div className="heading-container">
                                <img className="heading-image" src={azurelogo} />
                                <span className="heading-text">{Constants.MicrosoftAzureHeading}</span>
                            </div>
                            <div className="heading-app-text">
                                {Constants.GithubAppHeading}
                            </div>
                            {
                                this.state.mappingStateObject.successMessage != "" &&
                                <div className="successMessage">
                                    <img className="success-img" src={greentick_checkmark} width="20px" height="20px" />
                                    <span className="success-msg">{this.state.mappingStateObject.successMessage}</span>
                                </div>
                            }
                            {
                                this.state.mappingStateObject.error && this.state.mappingStateObject.errorMessage != "" &&
                                <div className="errorMessage">
                                    <img className="error-img" src={redcross} width="20px" height="20px" />
                                    <span className="error-msg">{Constants.InstallationMappingErrorMessage + " Error: " + this.state.mappingStateObject.errorMessage}</span>
                                </div>
                            }
                            {
                                this.state.mappingStateObject.error && this.state.mappingStateObject.errorMessage == "" &&
                                <div className="errorMessage">
                                    <img className="error-img" src={redcross} width="20px" height="20px" />
                                    <span className="error-msg">{Constants.InstallationMappingErrorMessage}</span>
                                </div>
                            }
                            <form
                                className="form-items"
                                onSubmit={(ev) => {
                                    ev.preventDefault();
                                    this.saveInstallationMapping();
                                }}
                            >
                                <Tenants {...tenantProps} />
                                <Subscriptions {...subscriptionProps}></Subscriptions>
                                <div className="subscription-link">
                                    <a className="link" href={Constants.CreateSubscriptionLink} rel="noopener noreferrer" target="_blank"> {Constants.CreateSubscription}</a>
                                </div>
                                <div className="description-link">
                                    {Constants.GithubAppDescription1}
                                    <a className="link" href={Constants.TermsOfServiceLink} rel="noopener noreferrer" target="_blank"> {Constants.TermsOfService}</a>, <a className="link bolt-link" href={Constants.PrivacyStatementLink} rel="noopener noreferrer" target="_blank">{Constants.PrivacyStatement}</a> and <a className="link bolt-link" href={Constants.CodeOfConductLink} rel="noopener noreferrer" target="_blank">
                                        {Constants.CodeOfConduct}
                                    </a>.
                                </div>
                            </form>
                        </div>

                        {
                            this.state.mappingStateObject.loading ?
                                <ClipLoader
                                    css={
                                        `display: block; float: right; margin-bottom: 30px; margin-right: 49px;`
                                    }
                                    size={40}
                                    loading={this.state.mappingStateObject.loading}
                                ></ClipLoader> :
                                <div className="save-button text-right">
                                    <button aria-disabled="false" className="bolt-button" data-focuszone="" data-is-focusable="true" type="submit" onClick={(ev) => {
                                        ev.preventDefault();
                                        this.saveInstallationMapping();
                                    }}>
                                        <span className="bolt-button-text">{Constants.SaveButtonText}</span>
                                    </button>
                                </div>
                        }
                    </div>
                </div>
            </div>
        )
    }
}

export default withAuthProvider(Mapping);