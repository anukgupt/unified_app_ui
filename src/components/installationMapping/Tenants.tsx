import React from 'react';
import { SelectMenu, Button } from '@primer/components';
import down from '../../images/downarrow.png';
import * as Constants from "../../Constants";
import { MappingStateObject } from './Mapping';
import '../../css/installationMapping/Mapping.css';
import redcross from '../../images/redcross.png';

export interface ITenantsProps {
  mappingStateObject: MappingStateObject,
  setTenantId: any
}

class Tenants extends React.Component<ITenantsProps> {
  render() {
    let tenants = this.props.mappingStateObject.tenants;
    var options = tenants && tenants.value &&
      tenants.value.length > 0 &&
      tenants.value.map(tenant =>
        <SelectMenu.Item className="formitem-selectmenu-item" onClick={() => {
          this.props.setTenantId(tenant);
        }} key={tenant.tenantId}>
          {tenant.displayName}
        </SelectMenu.Item>
      )
    return (
      <div className="formitem-tenant">
        <span>{Constants.SelectAzureTenantLabel}</span>
        <SelectMenu className="formitem-dropdown">
          <Button as="summary" className="formitem-button">
            <span className="formitem-button-value">{this.props.mappingStateObject.selectedTenant.name}</span>
            <img className="formitem-dropdown-arrow" alt="" src={down} />
          </Button>
          <SelectMenu.Modal width="370px">
            <SelectMenu.List>
              {options}
            </SelectMenu.List>
          </SelectMenu.Modal>
        </SelectMenu>
        {
          this.props.mappingStateObject.errorOnFetchingTenants &&
          <div className="errMessage">
            <img className="err-img" src={redcross} width="20px" height="20px" />
            <span className="err-msg">{Constants.ErrorOnFetchingTenantsMessage}</span>
          </div>
        }
      </div>
    );
  }
}

export default Tenants;