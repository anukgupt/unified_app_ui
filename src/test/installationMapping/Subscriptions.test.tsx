import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from "enzyme";
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Subscriptions, { ISubscriptionProps } from "../../components/installationMapping/Subscriptions";
import { getSubscriptions } from '../../services/installationMapping/subscriptionService';
import { basicURLS } from "../../config/Config";

Enzyme.configure({ adapter: new Adapter() });

const subscriptionProps: ISubscriptionProps = {
    getAccessToken: jest.fn(),
    setError: jest.fn(),
    tenantId: "tenant",
    setSubscriptionIds: jest.fn(),
    setAccessToken: jest.fn(),
    setErrorOnFetchingSubscriptions: jest.fn(),
    errorOnFetchingSubscriptions: false,
    errorOnFetchingTenants: false
}

test('renders subscriptions', () => {
    const subscriptionsUI = renderer.create(<Subscriptions {...subscriptionProps} />).toJSON();
    expect(subscriptionsUI).toMatchSnapshot()
});

test('getSubscriptions should fetch subscriptions', async () => {
    var subscriptions: any[] = ["default_1", "default_2"];
    const mockFetchPromise = Promise.resolve({
        json: () => Promise.resolve(subscriptions),
    });
    global.fetch = jest.fn().mockImplementation(() => mockFetchPromise);
    const subscriptionList = await getSubscriptions(basicURLS.fetchSubscriptionsURL, "");
    expect(global.fetch).toHaveBeenCalledTimes(1);
    expect(global.fetch).toHaveBeenCalledWith("https://management.azure.com/subscriptions?api-version=2020-01-01", { "headers": { "Authorization": "Bearer " }, "method": "GET" });
    expect(subscriptionList).toBeDefined();
    expect(subscriptionList).toStrictEqual(subscriptions);
});

test('test selectmenu', () => {
    const subscriptionsUI = mount(<Subscriptions {...subscriptionProps} />);
    expect(subscriptionsUI.find('.formitem-dropdown').exists()).toBeTruthy();
});

test('test selectmenu button', () => {
    const subscriptionsUI = mount(<Subscriptions {...subscriptionProps} />);
    expect(subscriptionsUI.find('.formitem-button').exists()).toBeTruthy();
});

test('test all subscriptions and a specific subscrption selectmenu item', () => {
    var subscriptions = [{ "subscriptionId": "subs1", "displayName": "subs1" }, { "subscriptionId": "subs2", "displayName": "subs2" }];
    const subscriptionsUI = mount(<Subscriptions {...subscriptionProps} />);
    subscriptionsUI.setState({ "subscriptionParameters": { "currentTenantId": "tenant", "filteredSubscriptions": subscriptions, "finalSubscriptionList": [], "searchValueForSubscriptions": "", "subscriptionRadioButtonValue": "All Subscriptions", "subscriptions": subscriptions } });
    expect(subscriptionsUI.find('#input-checkbox-all').exists).toBeTruthy();
    subscriptionsUI.find('#input-checkbox-all').simulate('click');
    expect(subscriptionsUI.state()).toStrictEqual({ "subscriptionParameters": { "currentTenantId": "tenant", "filteredSubscriptions": subscriptions, "finalSubscriptionList": [], "searchValueForSubscriptions": "", "subscriptionRadioButtonValue": "Select Subscriptions", "subscriptions": subscriptions } });
    expect(subscriptionsUI.find('#subs1').length).toBe(1);
});

