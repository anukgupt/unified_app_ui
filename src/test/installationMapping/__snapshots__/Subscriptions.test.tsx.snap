// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders subscriptions 1`] = `
<div
  className="formitem-subscription"
>
  <span>
    Select your Azure Subscription
  </span>
  <details
    className="SelectMenu__StyledSelectMenu-sc-12xd8o8-0 dxFRtv formitem-dropdown"
    onToggle={[Function]}
    open={false}
  >
    <summary
      className="ButtonBase-exshql-0 Button-sc-1cuu878-0 bdhCrA iJXmtE formitem-button"
      style={
        Object {
          "backgroundColor": "#fff",
          "border": "1px solid #8A8886",
          "borderRadius": "2px",
          "height": "24px",
          "padding": "0",
        }
      }
    >
      <span
        className="formitem-button-value"
      >
        Select Subscription
      </span>
      <img
        alt=""
        className="formitem-dropdown-arrow"
        src="downarrow.png"
      />
    </summary>
    <div
      className="SelectMenuModal__ModalWrapper-fv9zw3-1 dlCqxE"
      role="menu"
    >
      <div
        className="SelectMenuModal__Modal-fv9zw3-0 eBmamO"
        width="370px"
      >
        <form
          className="SelectMenuFilter__StyledForm-k2jey9-0 SCYOx"
        >
          <span
            className="TextInput__Wrapper-sc-156lfck-1 fUJJVy TextInput-wrapper"
            width="100%"
          >
            <input
              aria-label="Search for a subscription"
              className="TextInput__Input-sc-156lfck-0 cjybCM"
              onChange={[Function]}
              placeholder="Search for a subscription"
              type="text"
              value=""
            />
          </span>
        </form>
        <div
          className="SelectMenuList-uflcju-0 erwbXU"
        />
      </div>
    </div>
  </details>
  <div
    className="subscription-link"
  >
    <a
      className="link"
      href="https://go.microsoft.com/fwlink/?LinkID=266231"
      rel="noopener noreferrer"
      target="_blank"
    >
       
      Create subscription
    </a>
  </div>
</div>
`;
